src/data/reports/data-quality/DataQualityDefaultRepository.ts

```diff
@@ -0,0 +1,276 @@
+    }
+
+    async getProgramIndicators(
+        options: ProgramIndicatorOptions,
+        namespace: string
+    ): Promise<PaginatedObjects<ProgramIndicatorItem>> {
+        const dataQuality = await this.globalStorageClient.getObject<ProgramIndicatorConfig>(namespace);
+        const { paging, sorting } = options;
+
+        const dataQualityProgramIndicatorErrors =
+            dataQuality?.validationResults?.filter(
+                r => (!r.expressionresult || !r.filterresult) && r.metadataType === "ProgramIndicator"
+            ) ?? [];
+
+        const dataQualityIndicatorErrorsInPage = _(dataQualityProgramIndicatorErrors)
+            .orderBy([row => row[sorting.field]], [sorting.direction])
+            .drop((paging.page - 1) * paging.pageSize)
+            .take(paging.pageSize)
+            .value();
+
+        const pager: Pager = {
+            page: paging.page,
+            pageSize: paging.pageSize,
+            pageCount: Math.ceil(dataQualityProgramIndicatorErrors.length / paging.pageSize),
+            total: dataQualityProgramIndicatorErrors.length,
+        };
+        return { pager: pager, objects: dataQualityIndicatorErrorsInPage };
+    }
+
+    async saveDataQuality(namespace: string, dataQuality: IndicatorConfig | ProgramIndicatorConfig): Promise<void> {
+        return await this.globalStorageClient.saveObject<IndicatorConfig | ProgramIndicatorConfig>(
+            namespace,
+            dataQuality
+        );
+    }
+
+    async reloadValidation(namespace: string, fromZero: boolean) {
+        const dataQuality = await this.globalStorageClient.getObject<IndicatorConfig | ProgramIndicatorConfig>(
+            namespace
+        );
+        const dataQualityErrors: any[] = [];
+
+        if (fromZero && dataQuality?.validationResults.length === 0) {
+            const { indicators } = await this.api.metadata
+                .get({
+                    indicators: {
+                        fields: {
+                            id: true,
+                            name: true,
+                            numerator: true,
+                            denominator: true,
+                            lastUpdated: true,
+                            user: true,
+                        },
+                    },
+                })
+                .getData();
+            const { programIndicators } = await this.api.metadata
+                .get({
+                    programIndicators: {
+                        fields: {
+                            id: true,
+                            name: true,
+                            numerator: true,
+                            denominator: true,
+                            lastUpdated: true,
+                            user: true,
+                            expression: true,
+                            filter: true,
+                        },
+                    },
+                })
+                .getData();
```

Comment:
These and the ones from the `else if` below can be generated with a function with parameters to get this version or the one with filters. Makes the code more readable.

Reply:
programIndicators doesn't have  numerator or denominator, removed.